[NFC] Extend InjectTLIMappings pass testing

JolantaJensen opened the pull request:
This patch adds sleefgnuabi and ArmPL vector libraries to testing of InjectTLIMappings pass.

JolantaJensen committed:
[NFC] Extend InjectTLIMappings pass testing

This patch adds sleefgnuabi and ArmPL vector libraries to testing
of InjectTLIMappings pass.

mgabka commented on llvm/test/Transforms/Util/add-TLI-mappings.ll:
```diff
@@ -30,8 +29,12 @@ define double @sin_f64(double %in) {
+; ARMPL:        call double @sin(double %{{.*}})
 ; No mapping of "sin" to a vector function for Accelerate.
-; ACCELERATE-NOT: _ZGV_LLVM_{{.*}}_sin({{.*}})
+; ACCELERATE-NOT:  _ZGV_LLVM_{{.*}}_sin({{.*}})
```
Hi Jolanta,
Your changes to the test are wrong.
The purpose of this test is to check that the  "inject-tli-mappings" pass is adding the attribute to the function call with the mappings to vector variants of the scalar call  for both ARMPL and SLEEF such mappings exists so the check lines you added aren't correct.

You should also extend the check lines at the beginning and end of this file to show that the "llvm.compiler.used" array and "vector-function-abi-variant" attribute  look like when ArmPL and sleefgnuabi are used.

Perhaps the target needs to be passed as opt argument.

JolantaJensen replied:
Indeed, the target needed to be passed as argument. Fixed.

JolantaJensen committed:
Added -mtriple=aarch64-unknown-linux-gnu to the run lines for sleefgnuabi
and ArmPL as the TLI mappings for those libraries are only initialized
for aarch64. Corrected the check lines for sleefgnuabi and ArmPL.

mgabka commented on llvm/test/Transforms/Util/add-TLI-mappings.ll:
```diff
@@ -2,9 +2,8 @@
 ; RUN: opt -vector-library=MASSV -passes=inject-tli-mappings -S < %s | FileCheck %s  --check-prefixes=COMMON,MASSV
```
Despite the fact that MASSV, LIBMVEC or Accelerate libraries do not check if the target is correct, I think it is a good practice to specify it in the RUN line, otherwise it may lead to incorrect behaviour if something in the compiler change in the future.
So I think here we would need to add:  -mtriple=powerpc64-unknown-linux-gnu here

and  -mtriple=x86_64-unknown-linux-gnu in the X86 and Accelerate case

JolantaJensen committed:
Added target to the run lines that were missing it.
